redis/src/aio/async_transactions.rs

```diff
@@ -0,0 +1,127 @@
+use std::pin::Pin;
+
+use crate::{aio::ConnectionManager, cmd, pipe, Pipeline, RedisError, ToRedisArgs};
+use futures::Future;
+
+/// This function encapsulates the boilerplate required to establish a Redis transaction.
+/// Do not use it directly but use the `transaction_async!` macro instead.
+/// See the `transaction_async!` macro for more details.
+pub async fn transaction_async<
+    T,
+    E: From<RedisError>,
+    K: ToRedisArgs,
+    F: FnMut(
+        ConnectionManager,
+        &mut Pipeline,
+    ) -> Pin<Box<dyn Future<Output = Result<Option<T>, E>> + Send>>,
+>(
+    mut mgr: ConnectionManager,
+    keys: &[K],
+    mut func: F,
+) -> Result<T, E> {
+    loop {
+        cmd("WATCH").arg(keys).query_async(&mut mgr).await?;
+        let mut p = pipe();
+        let response: Option<T> = func(mgr.clone(), p.atomic()).await?;
```

Comment:
the `?` means that UNWATCH isn't called, correct?

The tests need to ensure that this leaves the connection in a clean state even when encountering an error.